source/adapters/level_zero/context.hpp

```diff
@@ -170,9 +170,9 @@ struct ur_context_handle_t_ : _ur_object {
 
   // Caches for events.
   using EventCache = std::vector<std::list<ur_event_handle_t>>;
-  EventCache EventCaches{4};
+  EventCache EventCaches{6};
```

Comment:
why 6? can you add some common place to control it ?

Reply:
6 is from the two extra slots given to counterbased immediate/regular event cache.

Reply:
we cannot have magic numbers , this is not maintainable

Reply:
@winstonzhang-intel, please add an enum value indicating the cache being targeted.